src/snowflake/snowpark/session.py

```diff
@@ -374,6 +376,8 @@ def create(self) -> "Session":
             """Creates a new Session."""
             if self._options.get("local_testing", False):
                 session = Session(MockServerConnection(self._options), self._options)
+                if "password" in self._options:
+                    self._options["password"] = None
```

Comment:
This is because we shouldn't store the password, see the live connection equivalent at: https://github.com/snowflakedb/snowpark-python/blob/7a42f03d6d3a60901b30453cfd01a3768fbdc483/src/snowflake/snowpark/_internal/server_connection.py#L159